kubernetes-docs/spbm.md

```diff
@@ -0,0 +1,196 @@
+ 
+ 
+**Note:** Here you don't need to create persistent volume it is created dynamically
+
+**Note:** VSAN storage capability based provisioning of persistent volumes is available in 1.6.5 release.
+
+
+*Create Storage Class*
+
+```
+vsphere-volume-sc-vsancapabilities.yaml
+
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: fast
+provisioner: kubernetes.io/vsphere-volume
+parameters:
+    diskformat: zeroedthick
+    hostFailuresToTolerate: "2"
+    cachereservation: "20"
+```
+
+Here a persistent volume will be created with the Virtual SAN capabilities - hostFailuresToTolerate to 2 and cachereservation is 20% read cache reserved for storage object. Also the persistent volume will be zeroedthickdisk.  
+
+The official VSAN policy documentation describes in detail about each of the individual storage capabilities that are supported by VSAN and can be configured on the virtual disk.
+You can also specify the datastore in the Storageclass as shown in above example. The volume will be created on the datastore specified in the storage class. This field is optional. If not specified as shown in example 1, the volume will be created on the datastore specified in the vsphere config file used to initialize the vSphere Cloud Provider.
+
+```
+vsphere-volume-sc-vsancapabilities.yaml
+
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: fast
+provisioner: kubernetes.io/vsphere-volume
+parameters:
+    diskformat: zeroedthick
+    datastore: VSANDatastore
+    hostFailuresToTolerate: "2"
+    cachereservation: "20"
+```
+
+**Note:** If you do not apply a storage policy during dynamic provisioning on a VSAN datastore, it will use a default Virtual SAN policy.
+
+**Create the storageclass**
+
+```
+$ kubectl create -f examples/volumes/vsphere/vsphere-volume-sc-vsancapabilities.yaml
+```
+
+**Verify storage class is created**
+
+```
+$ kubectl describe storageclass fast
+Name:		fast
+Annotations:	<none>
+Provisioner:	kubernetes.io/vsphere-volume
+Parameters:	diskformat=zeroedthick, hostFailuresToTolerate="2", cachereservation="20"
+No events.
+Create Persistent Volume Claim.
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: pvcsc-vsan
+  annotations:
+    volume.beta.kubernetes.io/storage-class: fast
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 2Gi
+ ```
+
+**Creating the persistent volume claim**
+
+```
+$ kubectl create -f examples/volumes/vsphere/vsphere-volume-pvcsc.yaml
+```
+
+**Verifying persistent volume claim is created**
+
+```
+$ kubectl describe pvc pvcsc-vsan
+Name:		pvcsc-vsan
+Namespace:	default
+Status:		Bound
+Volume:		pvc-80f7b5c1-94b6-11e6-a24f-005056a79d2d
+Labels:		<none>
+Capacity:	2Gi
+Access Modes:	RWO
+No events.
+Persistent Volume is automatically created and is bounded to this pvc.
```

Comment:
Seems to be mixing up content with the user output. Can you move it outside. This section as a whole doesn't look good.